### PR TITLE
Wouter load image

### PR DESCRIPTION
**Pull request by WoutervanGeel**
can load an image and return a list of pixels in Uint32 format

### CODE REVIEW

File: src/engine/SDLFacade.cpp [hunk 1]
```diff
@@ -179,7 +179,7 @@ namespace Engine {
 
         SDL_DestroyTexture(this->_screen_buffer);
 
-        this->_screen_buffer = SDL_CreateTexture(this->_renderer, SDL_PIXELFORMAT_ARGB8888, SDL_TEXTUREACCESS_STREAMING,
+        this->_screen_buffer = SDL_CreateTexture(this->_renderer, SDL_PIXELFORMAT_RGBA8888, SDL_TEXTUREACCESS_STREAMING,
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Annotate this somewhere (how was this enum found to be correct?)

File: src/engine/SDLFacade.cpp [hunk 2]
```diff
@@ -381,4 +381,36 @@ namespace Engine {
...
+    //todo: change given variable to match tilemaps
+    /// \brief Function that returns a list of pixels
+    ///
+    /// Each pixel in the list is represented as Uint32
+    /// \return This function returns a list of Uint32
+    vector<Uint32> SDLFacade::get_image_buffer(const string& path)
+    {
+        //todo: change given variable to match tilemaps
+        vector <Uint32> pixels;
+        const char* imagePath = path.c_str();
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: No need to do a variable assignment here, 
`SDL_LoadBMP(path.c_str())` is very readable and not at all confusing.

File: src/engine/SDLFacade.cpp [hunk 3]
```diff
@@ -381,4 +381,36 @@ namespace Engine {
...
+        SDL_Surface* image = SDL_LoadBMP(imagePath);
+        if(image == NULL){
+            cout << " NULL!" << endl;
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: Make this into a more descriptive error message. A random developer probably doesn't know where to look when he sees `NULL!`

File: src/engine/SDLFacade.cpp [hunk 4]
```diff
@@ -381,4 +381,36 @@ namespace Engine {
...
+            SDL_FreeSurface(image);
+            //todo: what to do when image can't be found
+            return pixels;
+        } else {
+            SDL_LockSurface(image);
+
+            for (int y = 0; y < image->h; y++) {
+                for (int x = 0; x < image->w; x++) {
+                    int bpp = image->format->BytesPerPixel;
```
Comment (ThomasvLingen) [inline src/engine/SDLFacade.cpp]: this assignment can be moved before the 2 for loops, saving us `image->h * image->w` variable assignments (in the case of a 64x64 image that's already 4096 variable assignments we save)

File: src/engine/SDLFacade.cpp (hunk 1 above)
Comment (WoutervanGeel) [inline src/engine/SDLFacade.cpp]: Done

File: src/engine/SDLFacade.cpp (hunk 2 above)
Comment (WoutervanGeel) [inline src/engine/SDLFacade.cpp]: goed punt. leek mij wat leesbaarder, maar als dat niet wordt ervaren, dan kan het gewoon weg:
Done

File: src/engine/SDLFacade.cpp (hunk 3 above)
Comment (WoutervanGeel) [inline src/engine/SDLFacade.cpp]: Done, goede feedback

File: src/engine/SDLFacade.cpp (hunk 4 above)
Comment (WoutervanGeel) [inline src/engine/SDLFacade.cpp]: thanks. niet gezien en erg goed. Done

**Pull request merged**